seatunnel-api/src/main/java/org/apache/seatunnel/api/table/catalog/CatalogTableUtil.java

```diff
@@ -126,8 +123,9 @@ public static List<CatalogTable> getCatalogTables(Config config, ClassLoader cla
                 .map(
                         catalog -> {
                             long startTime = System.currentTimeMillis();
-                            List<CatalogTable> catalogTables =
-                                    getCatalogTables(catalogConfig, catalog);
+                            catalog.open();
+                            List<CatalogTable> catalogTables = catalog.getTables(catalogConfig);
+                            catalog.close();
```

Comment:
If a program exception occurs before the `catalog.close()` method, will the connection be closed?